neo/Consensus/ConsensusService.cs

```diff
@@ -252,10 +281,74 @@ private void ExtendTimerByFactor(int maxDelayInBlockTimes)
                 ChangeTimer(nextDelay);
         }
 
-        private void OnConsensusPayload(ConsensusPayload payload)
+        private void TryToSaveFuturePayloads(ConsensusPayload payload)
+        {
+            // Limit the maximum future payload to be saved, 
+            // avoiding caching payload when node is lagged behind by more than furtherPayloadToSave locks
+            uint furtherPayloadToSave = 10;
+            if ((payload.BlockIndex >= context.Block.Index + 1) && (payload.BlockIndex <= context.Block.Index + furtherPayloadToSave))
+            {
+                ConsensusMessage futureMessage;
+                try
+                {
+                    futureMessage = payload.ConsensusMessage;
+                }
+                catch (FormatException)
+                {
+                    return;
+                }
+                catch (IOException)
+                {
+                    return;
+                }
+
+                switch (futureMessage)
+                {
+                    case ChangeView view:
+                        tryToSavePayloadIntoArray(context.FutureChangeViewPayloads, payload, futureMessage.ViewNumber);
+                        break;
+                    case Commit commit:
+                        tryToSavePayloadIntoArray(context.FutureCommitPayloads, payload, futureMessage.ViewNumber);
+                        break;
+                    case RecoveryMessage recovery:
+                        tryToSavePayloadIntoArray(context.FutureRecoveryPayloads, payload, futureMessage.ViewNumber);
+                        break;
+                    case PrepareRequest request:
+                    case PrepareResponse response:
+                        tryToSavePayloadIntoArray(context.FuturePreparationPayloads, payload, futureMessage.ViewNumber);
+                        break;
+                }
+            }
+        }
+
+        private void tryToSavePayloadIntoArray(ConsensusPayload[] payloadsArray, ConsensusPayload payload, byte pViewNumber)
```

Comment:
Lowercase method

Reply:
Fixed. 

About the main question, it is nothing similar to #933.